src/GeositeFramework/plugins/layer_selector_v2/main.js

```diff
@@ -160,14 +179,47 @@ define([
                     }
                 }, this);
 
-                _.each(visibleLayerIds, function(layerIds, serviceUrl) {
+                _.each(visibleLayerIds, function(layerServiceIds, serviceUrl) {
                     var mapLayer = this.map.getLayer(serviceUrl);
-                    if (layerIds.length === 0) {
+                    if (layerServiceIds.length === 0) {
                         mapLayer.setVisibleLayers([-1]);
                     } else {
-                        mapLayer.setVisibleLayers(layerIds);
+                        mapLayer.setVisibleLayers(layerServiceIds);
                     }
                 }, this);
+
+                this._setOpacityForService(visibleLayerIds);
+            },
+
+            _setOpacityForService: function(visibleLayerIds) {
+                // VisibleLayerIds is a map of serviceUrls -> selected layer service Ids
+                _.each(visibleLayerIds, function(layerServiceIds, serviceUrl) {
+                    var mapLayer = this.map.getLayer(serviceUrl),
+                        self = this;
+
+                    if (layerServiceIds.length !== 0) {
+                        var optionsArray = [],
+                            selectedLayers = this.state.getSelectedLayersForService(serviceUrl);
```

Comment:
This seems a little redundant. `layerServiceIds` should already contain all the selected layers for this service. Right?